app/services/request_approval_service.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class RequestApprovalService < BaseService
+  def call(user)
+    return true if user.id.nil?
+
+    if Setting.require_approval
+      User.admins.each do |admin|
+        UserMailer.new_user_waiting_for_approval(admin, user).deliver_later
+      end
+    else
+      user.approve
```

Comment:
Since this is a service object, I suggest doing the exclamation mark stuff so it raises an error, rather than relying on return values. So i.e. rename method to `approve!`, use `update!` from within